### DISCUSSION TITLE
Detokenize API endpoint

### DISCUSSION THREAD

**Discussion by konokonekonoko** [LostRuins/koboldcpp#622] [Q&A] [answered]
Does koboldcpp have a detokenize API endpoint or any way to split an input string up into its plaintext tokens?

**Comment by dan-1am** [comment, marked as answer]
API is described in built-in page: http://localhost:5001/api
/extra/tokencount - counts number of tokens and (in recent enough version) sends tokens ids of input text.

**Reply by konokonekonoko** [reply]
What I'm looking for is an option to turn the token ids back into text. But it doesn't look like it has an endpoint for that currently.
Thank you for your help.

**Reply by LostRuins** [reply]
A detokenization endpoint does not currently exist. What's the use case for it?

**Reply by konokonekonoko** [reply]
I'm implementing token and phrase biasing for mikupad, and I wanted to display the biased strings split into their tokens.

**Reply by LostRuins** [reply]
I see. I'll consider it for the future - it's not so straightforward due to multibyte token sequences.

**Reply by konokonekonoko** [reply]
Got it. Thanks for letting me know.
